src/llama_stack/providers/remote/prompts/mlflow/mlflow.py

```diff
@@ -0,0 +1,565 @@
+
+        Args:
+            prompt_id: Llama Stack prompt ID
+
+        Returns:
+            ListPromptsResponse with all versions of the prompt
+
+        Raises:
+            ValueError: If prompt not found
+        """
+        # MLflow doesn't have a direct "list versions" API for prompts
+        # We need to iterate and try to load each version
+        versions = []
+        version_num = 1
+        max_attempts = 100  # Safety limit
+
+        while version_num <= max_attempts:
+            try:
+                prompt = await self.get_prompt(prompt_id, version_num)
+                versions.append(prompt)
+                version_num += 1
+            except ValueError:
+                # No more versions
+                break
+            except Exception as e:
+                logger.warning(f"Error loading version {version_num} of {prompt_id}: {e}")
+                break
+
+        if not versions:
+            raise ValueError(f"Prompt {prompt_id} not found")
+
+        # Sort by version number
+        versions.sort(key=lambda p: p.version)
+
+        return ListPromptsResponse(data=versions)
+
+    async def set_default_version(self, prompt_id: str, version: int) -> Prompt:
+        """Set default version using MLflow alias.
+
+        Args:
+            prompt_id: Llama Stack prompt ID
+            version: Version number to set as default
+
+        Returns:
+            Prompt resource with is_default=True
+
+        Raises:
+            ValueError: If version not found
+        """
+        import mlflow
+
+        mlflow_name = self.mapper.to_mlflow_name(prompt_id)
+
+        # Verify version exists
+        try:
+            prompt = await self.get_prompt(prompt_id, version)
+        except ValueError as e:
+            raise ValueError(f"Cannot set default: {e}") from e
+
+        # Set "default" alias in MLflow
+        try:
+            mlflow.genai.set_prompt_alias(
+                name=mlflow_name,
+                version=version,
+                alias="default",
+            )
+            logger.info(f"Set version {version} as default for {prompt_id}")
+        except Exception as e:
+            logger.error(f"Failed to set default version: {e}")
+            raise
+
+        # Update is_default flag
+        prompt.is_default = True
+
+        return prompt
+
+    async def _is_default_version(self, mlflow_name: str, version: int) -> bool:
+        """Check if a version is the default version.
+
+        Args:
+            mlflow_name: MLflow prompt name
+            version: Version number
+
+        Returns:
+            True if this version is the default, False otherwise
+        """
+        import mlflow
+
+        try:
+            # Try to load with @default alias
+            default_uri = f"prompts:/{mlflow_name}@default"
+            default_prompt = mlflow.genai.load_prompt(default_uri)
+
+            # Get default version number
+            default_version = 1
```

Comment:
why do we have to set this?

Reply:
It first tries to load the alias `@default`; if that fails, it tries version `1`, which is the first version MLflow assigns to a prompt when it is registered for the first time without specifying an alias. More info https://mlflow.org/docs/latest/api_reference/python_api/mlflow.genai.html#mlflow.genai.set_prompt_alias

If it does not even have a version=1 registered, then the prompt definitely does not exist in the registry.